### ISSUE TITLE
Is this correct to write surrogate loss in this way?

### ISSUE THREAD

**Issue by TobyGE** [ajlangley/trpo-pytorch#3]
loss = self.surrogate_loss(log_action_probs, log_action_probs.detach(), advantages)

I can't tell the difference between log_action_probs and log_action_probs.detach(). I mean they have store same data, the only difference is gradient or not, so (log_action_probs - log_action_probs.detach()) = 0, which means the surrogate loss you compute is just torch.mean(advantages).

**Comment by ajlangley** [comment]
I'm sure you've long since answered your own question, but I thought I'd leave an answer here in case someone else is also uncertain on this point. 

Basically, there is a difference between evaluating the function at log_action_probs = log_action_probs.detach(), and evaluating its gradient at that point. In other words, even though the expression itself just becomes the mean of the advantages, the surrogate loss still yields useful gradient information to improve the policy.
